worlds/kh1/__init__.py

```diff
@@ -241,45 +345,260 @@ def set_rules(self):
+        keyblade_stats = VANILLA_KEYBLADE_STATS.copy()
+        # Handle shuffling keyblade stats
+        if self.options.keyblade_stats != "vanilla":
+            if self.options.keyblade_stats == "randomize":
+                # Fix any minimum and max values from settings
+                min_str_bonus = min(self.options.keyblade_min_str.value, self.options.keyblade_max_str.value)
+                max_str_bonus = max(self.options.keyblade_min_str.value, self.options.keyblade_max_str.value)
+                self.options.keyblade_min_str.value = min_str_bonus
+                self.options.keyblade_max_str.value = max_str_bonus
+                min_crit_rate = min(self.options.keyblade_min_crit_rate.value, self.options.keyblade_max_crit_rate.value)
+                max_crit_rate = max(self.options.keyblade_min_crit_rate.value, self.options.keyblade_max_crit_rate.value)
+                self.options.keyblade_min_crit_rate.value = min_crit_rate
+                self.options.keyblade_max_crit_rate.value = max_crit_rate
+                min_crit_str = min(self.options.keyblade_min_crit_str.value, self.options.keyblade_max_crit_str.value)
+                max_crit_str = max(self.options.keyblade_min_crit_str.value, self.options.keyblade_max_crit_str.value)
+                self.options.keyblade_min_crit_str.value = min_crit_str
+                self.options.keyblade_max_crit_str.value = max_crit_str
+                min_recoil = min(self.options.keyblade_min_recoil.value, self.options.keyblade_max_recoil.value)
+                max_recoil = max(self.options.keyblade_min_recoil.value, self.options.keyblade_max_recoil.value)
+                self.options.keyblade_min_recoil.value = min_recoil
+                self.options.keyblade_max_recoil.value = max_recoil
+                min_mp_bonus = min(self.options.keyblade_min_mp.value, self.options.keyblade_max_mp.value)
+                max_mp_bonus = max(self.options.keyblade_min_mp.value, self.options.keyblade_max_mp.value)
+                self.options.keyblade_min_mp.value = min_mp_bonus
+                self.options.keyblade_max_mp.value = max_mp_bonus
+                if self.options.bad_starting_weapons:
+                    starting_weapons = keyblade_stats[:4]
+                    other_weapons = keyblade_stats[4:]
+                else:
+                    starting_weapons = []
+                    other_weapons = keyblade_stats
+                for keyblade in other_weapons:
+                        keyblade["STR"] = int(self.random.randint(min_str_bonus, max_str_bonus))
+                        keyblade["CRR"] = int(self.random.randint(min_crit_rate, max_crit_rate))
+                        keyblade["CRB"] = int(self.random.randint(min_crit_str, max_crit_str))
+                        keyblade["REC"] = int(self.random.randint(min_recoil, max_recoil))
+                        keyblade["MP"]  = int(self.random.randint(min_mp_bonus, max_mp_bonus))
```

Comment:
```suggestion
                        keyblade["STR"] = self.random.randint(min_str_bonus, max_str_bonus)
                        keyblade["CRR"] = self.random.randint(min_crit_rate, max_crit_rate)
                        keyblade["CRB"] = self.random.randint(min_crit_str, max_crit_str)
                        keyblade["REC"] = self.random.randint(min_recoil, max_recoil)
                        keyblade["MP"]  = self.random.randint(min_mp_bonus, max_mp_bonus)
```
`random.randint` returns an int already